### ISSUE TITLE
FixedChannelPool does not properly counts the acquiredChannelCount

### ISSUE THREAD

**Issue by ioanbsu** [netty/netty#4023]
Steps to reproduce:
1) create FixedChannelPool with maxConnections=1
```java
FixedChannelPool(bootstrap.remoteAddress(key.host(), key.port()),
 new AbstractChannelPoolHandler() {... },1);
```
2) Request channel from FixedChannelPool
3) Make request through channel and once done release it
4) Kill the connection while channel is in pool.
5) Request channel from the pool. Since connection is not healthy anymore it will close the channel and right after that will try to acquire new channel. But acquiredChannelCount is never decremented. 
So at the point when FixedChannelPool will try to acquire connection it will think that acquiredChannelCount==1 and it won't force new connection creation, instead it will put task to the pendingAcquireQueue, see lines 195 to 226 in FixedChannelPool.

Expected result:
When connection is died(becomes unhealthy) while being in pool it should recognize that in FixedChannlePool and decrement acquiredChannelCount before asking for new channel creation.
Actual result:
acquiredChannelCount stays unchanged so new channel never gets created/acquired.

Suggested solution:
in SimpleChannelPool add code:
```java
  private void onChannelUnhealthy(Promise<Void> channelClosePromise) {
        channelClosePromise.addListener(new GenericFutureListener<Future<? super Void>>() {
            @Override
            public void operationComplete(Future<? super Void> future) throws Exception {
                channelClosedCauseUnhealthy();
            }
        });
    }

    protected void channelClosedCauseUnhealthy() {

    }
```
and modify notifyHealthCheck(...) to be as following:
```java
 private void notifyHealthCheck(Future<Boolean> future, Channel ch, Promise<Channel> promise) {
        assert ch.eventLoop().inEventLoop();

        if (future.isSuccess()) {
            if (future.getNow() == Boolean.TRUE) {
                try {
                    ch.attr(POOL_KEY).set(this);
                    handler.channelAcquired(ch);
                    promise.setSuccess(ch);
                } catch (Throwable cause) {
                    Promise<Void> channelClosePromise = ch.eventLoop().<Void>newPromise();
                    onUnhealthyChannelClosed(channelClosePromise);
                    closeAndFail(ch, cause, promise,channelClosePromise);
                }
            } else {
                Promise<Void> channelClosePromise = ch.eventLoop().<Void>newPromise();
                onUnhealthyChannelClosed(channelClosePromise);
                closeChannel(ch, channelClosePromise);
                acquire(promise);
            }
        } else {
            Promise<Void> channelClosePromise = ch.eventLoop().<Void>newPromise();
            onUnhealthyChannelClosed(channelClosePromise);
            closeChannel(ch, channelClosePromise);
            acquire(promise);
        }
    }
```
closeChannel function needs to be modified as well:
```java
 private static void closeChannel(final Channel channel, final Promise channelClosePromise) {
        channel.attr(POOL_KEY).getAndSet(null);
        ChannelFuture future = channel.close();
        if (channelClosePromise != null) {
            future.addListener(new GenericFutureListener<Future<? super Void>>() {
                @Override
                public void operationComplete(Future<? super Void> future) throws Exception {
                    channelClosePromise.setSuccess(future.isSuccess());
                }
            });
        }
    }
  private static void closeAndFail(Channel channel, Throwable cause, Promise<?> promise,
                                     Promise<Void> channelClosePromise) {
        closeChannel(channel, channelClosePromise);
        promise.setFailure(cause);
    }
```
in FixedChannelPool add method:
```java
   protected void channelClosedCauseUnhealthy() {
       decrementAndRunTaskQueue();
    }
```

**Comment by ioanbsu** [comment]
Please review pull request: https://github.com/netty/netty/pull/4027

**Comment by normanmaurer** [comment]
Will do tomorrow!

**Comment by ioanbsu** [comment]
Applied all the changes to satisfy #4027 pull requests comments.
Unfortunately I somehow closed my original pull request. Probably because I tried to do force push so my commits are squashed.. 

I had to create new pull request that contains all my original changes and changes that were suggested by @normanmaurer  in my original pull request.
Please review https://github.com/netty/netty/pull/4029

**Comment by normanmaurer** [comment]
Fixed by #4076
